content/en/docs/ops/common-problems/network-issues/index.md

```diff
@@ -233,6 +233,112 @@ server {
 }
 {{< /text >}}
 
+## 503 error while accessing headless services
+
+Assume Istio is installed with the following configuration:
+
+- `mTLS mode` set to `STRICT` within the mesh
+- `meshConfig.outboundTrafficPolicy.mode` set to `ALLOW_ANY`
+
+Consider `nginx` is deployed as a `StatefulSet` in the default namespace and a corresponding `Headless Service` is defined as shown below:
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  labels:
+    app: nginx
+spec:
+  ports:
+  - port: 80
+    name: http-web  # Explicitly defining an http port
+  clusterIP: None   # Creates a Headless Service
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+The port name `http-web` in the Service definition explicitly specifies the http protocol for that port.
+
+Let us assume we have a [sleep]({{< github_tree >}}/samples/sleep) pod `Deployment` as well in the default namespace.
+When `nginx` is accessed from this `sleep` pod using its Pod IP (this is one of the common ways to access a headless service), the request goes via the `PassthroughCluster` to the server-side, but the sidecar proxy on the server-side fails to find the route entry to `nginx` and fails with `HTTP 503 UC`.
+
+{{< text bash >}}
+$ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath='{.items..metadata.name}')
+$ kubectl exec -it $SOURCE_POD -c sleep -- sh
+/ $ curl 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+  503/
```

Comment:
This doesn't display properly in preview and won't work with cut-paste. How about calling curl directly?
```suggestion
$ kubectl exec -it $SOURCE_POD -c sleep -- curl 10.1.1.171 -s -o /dev/null -w "%{http_code}"
503
```

Reply:
I thought about this too, but the problem is Pod IP (10.1.1.171 here) is not a constant. When this command is copy-pasted from the doc page, wrong Pod IP would be copied always and the user will have to then manually change it. If you keep it the way it is now, copy-paste will exec into the sleep container and wait for the user's curl <Pod IP> input.

Reply:
Yes, but the problem with the way you have it is that the copy button doesn't copy any of the curl command (and it doesn't display so nicely either - see [preview](https://deploy-preview-10071--preliminary-istio.netlify.app/latest/docs/ops/common-problems/network-issues/#503-error-while-accessing-headless-services)) so the user needs to type the whole line. In other similar docs the copy will pick up the whole command so that after the paste, the user can just hit the `back-arrow` key and edit the IP before hitting `enter`.

In other similar docs we either mention that `10,.1.1.171` is the IP of one of the pods or we document it as `curl <pod ip> -s -o /dev/null -w "%{http_code}"` and tell the user to change `<pod ip>` to the IP of one of the pods.

Reply:
Thanks for the explanation!
I have pushed all these changes in a new commit.